File: app/src/main/java/com/lavish/toprestro/models/Review.java
package com.lavish.toprestro.models;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

public class Review {

    public String id, userName, review, reply, ownerEmail;
    public Date timestamp;
    public int starRating;

    public Review(String userName, String ownerEmail, String review, int starRating) {
        this.userName = userName;
        this.review = review;
        this.ownerEmail = ownerEmail;
        this.starRating = starRating;

        timestamp = Calendar.getInstance().getTime();
    }

    public String formattedDate(){
        SimpleDateFormat sdf = new SimpleDateFormat("dd MMM", Locale.ENGLISH);
        return sdf.format(timestamp);
    }

}

File: app/src/main/java/com/lavish/toprestro/featureOwner/domain/model/Review.kt
package com.lavish.toprestro.featureOwner.domain.model

import com.lavish.compose.ui.theme.Stars2
import com.lavish.compose.ui.theme.Stars3
import com.lavish.compose.ui.theme.Stars4
import com.lavish.compose.ui.theme.Stars5
import java.text.SimpleDateFormat
import java.util.*

data class Review(var id: String?,
        var timestamp: Date,

        var userName: String,
        var userEmail: String,

        var ownerEmail: String,
        var restaurantId: String,

        var starRating: Float,
        var review: String,
        var reply: String?,
        var restroName: String) {

    constructor(): this(
        "",
        Date(),
        "",
        "",
        "",
        "",
        0f,
        "",
        "",
        ""
    )

    fun formattedDate(): String {
        val sdf = SimpleDateFormat("dd MMM", Locale.ENGLISH)
        return sdf.format(timestamp)
    }

    fun colorForRating() = when {
        starRating > 4 -> Stars5
        starRating > 3 -> Stars4
        starRating > 2 -> Stars3
        else -> Stars2
    }


}